test/suites/dev-tanssi-relay/services-payment/test_services_payment_block_credits.ts

```diff
@@ -0,0 +1,275 @@
+                    const collators = await polkadotJs.query.tanssiCollatorAssignment.collatorContainerChain();
+
+                    // We are evaluating blockCredits for now, so lets put a lot of collatorAssignmentCredits
+                    const tx = polkadotJs.tx.servicesPayment.setCollatorAssignmentCredits(paraId, 1000n);
+                    await context.createBlock([await polkadotJs.tx.sudo.sudo(tx).signAsync(alice)]);
+
+                    // Container chain 2001 does not have any collators, this will result in only 1 container chain
+                    // producing blocks at a time. So if both container chains have 1000 credits, container 2000
+                    // will produce blocks 0-999, and container 2001 will produce blocks 1000-1999.
+                    if (paraId == 2000) {
+                        expect(
+                            collators.toJSON().containerChains[paraId].length,
+                            `Container chain ${paraId} has 0 collators`
+                        ).toBeGreaterThan(0);
+                    }
+                }
+            },
+        });
+
+        it({
+            id: "E02",
+            title: "Creating a container chain block costs credits",
+            test: async function () {
+                // Read num credits of para 2000, then create that many blocks. Check that authorNoting.blockNum does not increase anymore
+                // and collatorAssignment does not have collators
+
+                // create at least a couple blocks to at least see one block being consumed
+                // we will be doing this for the whole test, i.e., creating two blocks to ensure the parachain advances
+                await context.createBlock();
+                await context.createBlock();
+
+                const paraId = 2000n;
+
+                // Create a block, the block number should increase, and the number of credits should decrease
+                const credits1 = (await polkadotJs.query.servicesPayment.blockProductionCredits(paraId)).toJSON();
+                const containerBlockNum1 = await (await polkadotJs.query.authorNoting.latestAuthor(paraId)).toJSON()
+                    .blockNumber;
+
+                // create at least a couple blocks to at least see one block being consumed
+                await context.createBlock();
+                await context.createBlock();
+
+                const credits2 = (await polkadotJs.query.servicesPayment.blockProductionCredits(paraId)).toJSON();
+                const containerBlockNum2 = await (await polkadotJs.query.authorNoting.latestAuthor(paraId)).toJSON()
+                    .blockNumber;
+                expect(containerBlockNum1, "container chain 2000 did not create a block").toBeLessThan(
+                    containerBlockNum2
+                );
+                expect(credits1, "container chain 2000 created a block without burning any credits").toBeGreaterThan(
+                    credits2
+                );
```

Comment:
Can we also test that `credits1 - credits2 == containerBlockNum2 - containerBlockNum1`? I'm assuming that 1 block created is always 1 credit

Reply:
done